packages/fxa-shared/metrics/dnt.ts

```diff
@@ -0,0 +1,19 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const denyPrefixes = ['entrypoint', 'utm'];
+const denyNames = ['product_id', 'plan_id', 'productId', 'planId'];
+
+const startsWithPrefix = (key: string) =>
+  denyPrefixes.some((prefix) => key.startsWith(prefix));
+
+export const filterDntValues = (metricsData: { [keys: string]: unknown }) =>
+  Object.entries(metricsData).reduce((acc, [k, v]) => {
```

Comment:
Since we call this from JS code where type safety isn't garuanteed, we should probably do some null|undefined checks here.

Reply:
Did you mean null|undefined check on `metricsData`?

Reply:
Yes, this is what I am thinking. It's probably not entirely necessary. After all, automated and manual testing didn't uncover any issues. However, I noticed places where `filterDntValues` was called and the data provided seemed a bit contextual. For example, if you look at content server, it wasn't super obvious to me where something like event_properties or user_properties was checked / enforced.

Reply:
I think it's ok.  If a null or undefined is passed in, we'd learn about it in Sentry (if not caught during development).